packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -0,0 +1,678 @@
+                uniform vec4 grl_dashOptions;
+                uniform vec4 grl_colorMode_visibility_colorsWidth_useColors;
+                uniform vec3 grl_singleColor;
+                `,
+        };
+    }
+
+    // only getter, it doesn't make sense to use this plugin on a mesh other than GreasedLineMesh
+    // and it doesn't make sense to disable it on the mesh
+    get isEnabled() {
+        return true;
+    }
+
+    /**
+     * Bind the uniform buffer
+     * @param uniformBuffer
+     */
+    bindForSubMesh(uniformBuffer: UniformBuffer) {
+        const activeCamera = this._scene.activeCamera;
+
+        if (activeCamera) {
+            const projection = activeCamera.getProjectionMatrix();
+            uniformBuffer.updateMatrix("grl_projection", projection);
+        } else {
+            throw Error("GreasedLinePluginMaterial requires an active camera.");
+        }
+
+        const resolutionLineWidth = TmpVectors.Vector3[0];
+        if (this._options.resolution) {
+            resolutionLineWidth.x = this._options.resolution.x;
+            resolutionLineWidth.y = this._options.resolution.y;
+        } else {
+            resolutionLineWidth.x = this._engine.getRenderWidth();
+            resolutionLineWidth.y = this._engine.getRenderHeight();
+        }
+        resolutionLineWidth.z = this._options.width ? this._options.width : this._options.sizeAttenuation ? 1 : 0.1;
+        uniformBuffer.updateVector3("grl_resolution_lineWidth", resolutionLineWidth);
+
+        const dashOptions = TmpVectors.Vector4[0];
+        dashOptions.x = GreasedLinePluginMaterial._BooleanToNumber(this._options.useDash);
+        dashOptions.y = this._dashArray ?? 0;
+        dashOptions.z = this._options.dashOffset ?? 0;
+        dashOptions.w = this._options.dashRatio ?? 0.5;
+        uniformBuffer.updateVector4("grl_dashOptions", dashOptions);
+
+        const colorModeVisibilityColorsWidthUseColors = TmpVectors.Vector4[1];
+        colorModeVisibilityColorsWidthUseColors.x = this._options.colorMode ?? GreasedLineMeshColorMode.COLOR_MODE_SET;
+        colorModeVisibilityColorsWidthUseColors.y = this._options.visibility ?? 1;
+        colorModeVisibilityColorsWidthUseColors.z = this._colorsTexture ? this._colorsTexture.getSize().width : 0;
+        colorModeVisibilityColorsWidthUseColors.w = GreasedLinePluginMaterial._BooleanToNumber(this._options.useColors);
+        uniformBuffer.updateVector4("grl_colorMode_visibility_colorsWidth_useColors", colorModeVisibilityColorsWidthUseColors);
+
+        if (this._options.color) {
+            uniformBuffer.updateColor3("grl_singleColor", this._options.color ?? Color3.White());
+        }
+
+        if (this._colorsTexture) {
+            uniformBuffer.setTexture("grl_colors", this._colorsTexture);
+        } else if (this._engine.isWebGPU) {
+            uniformBuffer.setTexture("grl_colors", GreasedLinePluginMaterial._EmptyColorsTexture);
+        }
+
+        uniformBuffer.update();

```

Comment:
```suggestion
```

Reply:
Thanks!

> the sampler2D is used only when grlUseColors is set to 1 and in this case it is guaranteed that there will be the correct texture set.

What I commented on was the other case, when `grlUseColors=0`.

In this case, the `sampler2D` is not used in a texture lookup (so there won't be any rendering bugs), but it's still declared in the code. In this case, WebGL (maybe it's us, maybe it's Angle, I don't know) will still bind a texture, but a random texture, which is why I want to avoid this happening, because sometimes it can make a debugging session more difficult (when using Spector for example).